### PR TITLE
Use async `check` instead of the sync one

### LINKED ISSUE #1725: Docs: Replace sync `check` usage with async `check`
Change all, from:

```js
const headerText = await page.locator('h2').textContent();
check(headerText, {
  'header': headerText === 'Welcome, admin!'
});
```

To:

```js
import { check } from 'https://jslib.k6.io/k6-utils/1.5.0/index.js';

// ...

check(page.locator('h2'), {
  'header': async lo => await lo.textContent() === 'Welcome, admin!'
});
```

```[tasklist]
### Tasks
- [ ] grafana/k6-docs#1746
- [ ] grafana/xk6-browser#1448
```

### PR DESCRIPTION
**Pull request by inancgumus**
## What?

We're replacing sync `check` of k6 with the async `check` utility function. Along the way, I've also fixed some small issues I saw.

## Why

Writing concise code can be difficult when using [the k6 `check` function](https://grafana.com/docs/k6/latest/javascript-api/k6/check/) with async code since it doesn't support async APIs. A solution that we have suggested so far is to declare a temporary variable, wait for the value that is to be checked, and then check the result later. However, this approach can clutter the code with single-use declarations and unnecessary variable names, for example:

```javascript
const checked = await p.locator('.checked').isChecked();

check(checked, {
    'checked': c => c,
});
```

To address this limitation, we've added a version of the `check` function to [jslib.k6.io](https://jslib.k6.io/) that makes working with `async`/`await` simpler. The `check` function is a drop-in replacement for the built-in check, with added support for async code. Any `Promise`s will be awaited, and the result is reported once the operation has been completed:

```javascript
// Import the new check function from jslib.k6.io/k6-utils
import { check } from 'https://jslib.k6.io/k6-utils/1.5.0/index.js';

// ...

// Use the new check function with async code
check(page, {
    'checked': async p => p.locator('.checked').isChecked(),
});
```

## Related PR(s)/Issue(s)

Closes #1725

## Checklist

- [x] I have used a meaningful title for the PR.
- [x] I have described the changes I've made in the "What?" section above.
- [x] I have performed a self-review of my changes.
- [x] I have run the `npm start` command locally and verified that the changes look good.

- [x] I have made my changes in the `docs/sources/next` folder of the documentation.
- [x] I have reflected my changes in the `docs/sources/v{most_recent_release}` folder of the documentation.

### CODE REVIEW

File: docs/sources/next/using-k6-browser/migrating-to-k6-v0-52.md [hunk 1]
```diff
@@ -275,7 +275,7 @@ Below is a screenshot of a comparison between a generic browser test in `v0.51`
 
 ## Working with k6 check
 
-The k6 `check` API will not `await` promises, so calling a function that returns a `Promise`, for example `locator.textContent()`, inside one of the predicates will not work. Instead you will have to `await` and store the result in a variable _outside_ the `check`:
+The k6 `check` API will not `await` promises, so calling a function that returns a `Promise`, for example `locator.textContent()`, inside one of the predicates will not work. Instead you can work with [the jslib.k6.io's `check` utility](https://grafana.com/docs/k6/latest/javascript-api/jslib/utils/check/).
```
Comment (heitortsergent) [inline docs/sources/next/using-k6-browser/migrating-to-k6-v0-52.md:278]:
```suggestion
The k6 `check` API will not `await` promises, so calling a function that returns a `Promise`, for example `locator.textContent()`, inside one of the predicates will not work. Instead, you can work with [the jslib.k6.io's `check` utility function](https://grafana.com/docs/k6/latest/javascript-api/jslib/utils/check/).
```

**Comment by heitortsergent** [comment]
The changes look good to me, I'll wait until someone else on the development team takes a look. 🤓 

Also, good catch on the `k6/x/browser` import. I did a search and found a few more instances of it (throttlenetwork.md, on.md, waitforevent.md). Do you mind if I make a commit to fix those as well?

**Comment by inancgumus** [comment]
@heitortsergent Thanks for your review!

> The changes look good to me, I'll wait until someone else on the development team takes a look. 🤓

❤️ Of course!

> Also, good catch on the k6/x/browser import. I did a search and found a few more instances of it (throttlenetwork.md, on.md, waitforevent.md). Do you mind if I make a commit to fix those as well?

Thanks! Of course, please feel free :)

File: docs/sources/next/using-k6-browser/migrating-to-k6-v0-52.md (hunk 1 above)
Comment (inancgumus) [reply docs/sources/next/using-k6-browser/migrating-to-k6-v0-52.md]: I've also updated the previous versions: [3c22591](https://github.com/grafana/k6-docs/pull/1746/commits/3c22591877f68479868f3c459da1b7eda1d4208a)

**Comment by inancgumus** [comment]
@heitortsergent In addition to your commit, I've also [added another]([b977c76](https://github.com/grafana/k6-docs/pull/1746/commits/b977c76b700a1c5a65c4ee8ee7ad9c413fd72045)) that fixes the ones in v51 and below.

File: docs/sources/v0.52.x/javascript-api/k6-browser/page/waitforfunction.md [hunk 2]
```diff
@@ -64,8 +64,10 @@ export default async function () {
       polling: 'mutation',
       timeout: 2000,
     });
-    const innerHTML = await ok.innerHTML();
-    check(ok, { 'waitForFunction successfully resolved': innerHTML == 'Hello' });
+    await check(page, {
+      'waitForFunction successfully resolved': async () =>
+          await ok.innerHTML() == 'Hello'
+    });
```
Comment (ankur22) [inline docs/sources/v0.52.x/javascript-api/k6-browser/page/waitforfunction.md:70]:
```suggestion
    await check(ok, {
      'waitForFunction successfully resolved': async (ok) =>
          await ok.innerHTML() == 'Hello'
    });
```

File: docs/sources/v0.53.x/javascript-api/k6-browser/page/waitforfunction.md [hunk 3]
```diff
@@ -64,8 +64,10 @@ export default async function () {
       polling: 'mutation',
       timeout: 2000,
     });
-    const innerHTML = await ok.innerHTML();
-    check(ok, { 'waitForFunction successfully resolved': innerHTML == 'Hello' });
+    await check(page, {
+      'waitForFunction successfully resolved': async () =>
+          await ok.innerHTML() == 'Hello'
+    });
```
Comment (ankur22) [inline docs/sources/v0.53.x/javascript-api/k6-browser/page/waitforfunction.md:70]:
```suggestion
    await check(ok, {
      'waitForFunction successfully resolved': async (ok) =>
          await ok.innerHTML() == 'Hello'
    });
```

File: docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md [hunk 4]
```diff
@@ -132,10 +130,10 @@ To find out more information about injecting faults to your service, check out t
 {{< code >}}
 
 ```javascript
-import http from 'k6/http';
-import { check } from 'k6';
 import { browser } from 'k6/browser';
+import { check } from 'https://jslib.k6.io/k6-utils/1.5.0/index.js';
 import { ServiceDisruptor } from 'k6/x/disruptor';
+import http from 'k6/http';
```
Comment (ankur22) [inline docs/sources/next/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md:136]: Doesn't look like we need this import.

File: docs/sources/v0.52.x/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md [hunk 5]
```diff
@@ -188,9 +186,8 @@ export async function checkFrontend() {
 
   try {
     await page.goto(BASE_URL);
-    const header = await page.locator('h1').textContent();
-    check(header, {
-      header: (h) => h == 'Looking to break out of your pizza routine?',
+    check(page.locator('h1'), {
```
Comment (ankur22) [inline docs/sources/v0.52.x/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md:189]: The import should be `import { check } from 'https://jslib.k6.io/k6-utils/1.5.0/index.js';`. This example is still using `k6` `check`.

File: docs/sources/v0.53.x/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md [hunk 6]
```diff
@@ -188,9 +186,8 @@ export async function checkFrontend() {
 
   try {
     await page.goto(BASE_URL);
-    const header = await page.locator('h1').textContent();
-    check(header, {
-      header: (h) => h == 'Looking to break out of your pizza routine?',
+    check(page.locator('h1'), {
```
Comment (ankur22) [inline docs/sources/v0.53.x/using-k6-browser/recommended-practices/hybrid-approach-to-performance.md:189]: The import should be `import { check } from 'https://jslib.k6.io/k6-utils/1.5.0/index.js';`. This example is still using `k6` `check`.

**Review by ankur22 — APPROVED** [review]
Thanks for this change 😄 

Left some suggestions that might need to be looked at.

**Review by andrewslotin — APPROVED** [review]

**Comment by inancgumus** [comment]
@ankur22 Thanks for your review! I've made the updates 👍

**Review by heitortsergent — APPROVED** [review]

**Review by heitortsergent — APPROVED** [review]

**Review by ankur22 — APPROVED** [review]

**Pull request merged**